pacman/utilities/json_utils.py

```diff
@@ -283,57 +335,125 @@ def vertex_add_contstraints_from_json(json_dict, graph):
+        json_list.append(edge_to_json(edge))
+    json_dict["edges"] = json_list
+    return json_dict
+
+
+def partition_from_json(json_dict, graph):
+    identifier = json_dict["identifier"]
+    pre_vertex = vertex_lookup(json_dict["pre_vertex"], graph)
+    for j_edge in json_dict["edges"]:
+        edge = edge_from_json(j_edge, pre_vertex, graph)
+        graph.add_edge(edge, identifier)
 
 
-def graph_to_json(graph):
+def machine_graph_to_json(graph):
     json_dict = OrderedDict()
     try:
         if graph.label is not None:
             json_dict["label"] = graph.label
         json_list = []
         for vertex in graph.vertices:
-            json_list.append(vertex_to_json(vertex))
+            json_list.append(machine_vertex_to_json(vertex))
         json_dict["vertices"] = json_list
         json_list = []
-        for edge in graph.edges:
-            json_list.append(edge_to_json(edge))
-        json_dict["edges"] = json_list
+        for partition in graph.outgoing_edge_partitions:
+            json_list.append(partition_to_json(partition))
+        json_dict["partitions"] = json_list
     except Exception as ex:  # pylint: disable=broad-except
         json_dict["exception"] = str(ex)
     return json_dict
 
 
-def graph_from_json(json_dict):
+def machine_graph_from_json(json_dict):
     json_dict = json_to_object(json_dict)
     graph = MachineGraph(json_dict.get("label"))
     for j_vertex in json_dict["vertices"]:
-        graph.add_vertex(vertex_from_json(j_vertex, convert_constraints=False))
+        graph.add_vertex(machine_vertex_from_json(
+            j_vertex, convert_constraints=False))
     # Only do constraints when we have all the vertexes to link to
     for j_vertex in json_dict["vertices"]:
         vertex_add_contstraints_from_json(j_vertex, graph)
-    for j_edge in json_dict["edges"]:
-        edge = edge_from_json(j_edge, graph)
-        graph.add_edge(edge, "JSON_MOCK")
+    for j_partitions in json_dict["partitions"]:
+        partition_from_json(j_partitions, graph)
     return graph
 
 
+def graphs_to_json(application_graph, machine_graph, graph_mapper):
+    """
+
+    :param ApplicationGraph application_graph:
+    :param MachineGraph machine_graph:
+    :param graph_mapper:  TODO NUKE ME!
+    :return:
+    """
+    json_dict = OrderedDict()
+    try:
+        if machine_graph.label is not None:
+            json_dict["label"] = machine_graph.label
+        json_list = []
+        for application_vertex in application_graph.vertices:
+            json_list.append(application_vertex_to_json(
+                application_vertex, graph_mapper))
+        json_dict["vertices"] = json_list
+        json_list = []
+        for partition in machine_graph.outgoing_edge_partitions:
+            json_list.append(partition_to_json(partition))
+        json_dict["partitions"] = json_list
+    except Exception as ex:  # pylint: disable=broad-except
+        json_dict["exception"] = str(ex)
+    return json_dict
```

Comment:
This seems to take the vertices from the application graph but the edges from the machine graph.  Is this what you wanted?

Reply:
Yes The only role of the Application vertexes is to group the Machine Vertexes by Application vertex so that the key Allocator and ensure the keys for two different Application Vertexes do not intersect.